### ISSUE TITLE
Gallery: Add inline editing to allow removing images from block

### ISSUE THREAD

**Issue by mkaz** [WordPress/gutenberg#2128]
The gallery block currently relies on the core media modal dialog to remove individual images, this should be managed within the block itself. 

Add (x) for each image allowing for it to be removed from gallery block
